os/binfmt/libelf/libelf_bind.c

```diff
@@ -194,32 +195,40 @@ static int elf_relocate(FAR struct elf_loadinfo_s *loadinfo, int relidx, FAR con
 
 	for (i = 0; i < relsec->sh_size / sizeof(Elf32_Rel); i++) {
 		psym = &sym;
+		prel = &rel;
+
 
 		/* Read the relocation entry into memory */
 
-		ret = elf_readrel(loadinfo, relsec, i, &rel);
-		if (ret < 0) {
-			berr("Section %d reloc %d: Failed to read relocation entry: %d\n", relidx, i, ret);
-			goto ret_err;
+		if (loadinfo->reltab) {
+			prel = loadinfo->reltab + sizeof(Elf32_Rel) * i;
+		} else {
+			ret = elf_readrel(loadinfo, relsec, i, &rel);
```

Comment:
In elf_readrel(), remove if (loadinfo->reltab)  portion.

Reply:
Done. Please check.